<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>910</width>
    <height>660</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QPushButton" name="clear_restrict_functions">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>440</y>
     <width>180</width>
     <height>28</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>180</width>
     <height>0</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>180</width>
     <height>16777215</height>
    </size>
   </property>
   <property name="text">
    <string>Clear all restrict functions</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="line_insert_objective_function">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>40</y>
     <width>550</width>
     <height>30</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>550</width>
     <height>30</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>550</width>
     <height>30</height>
    </size>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>360</y>
     <width>300</width>
     <height>25</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>300</width>
     <height>25</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>300</width>
     <height>25</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Insert restrict function as:  g(x[n]) &lt;= 0</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="line_insert_restrict_function">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>400</y>
     <width>300</width>
     <height>30</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>300</width>
     <height>30</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>300</width>
     <height>30</height>
    </size>
   </property>
  </widget>
  <widget class="QPushButton" name="add_restrict_function">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>440</y>
     <width>180</width>
     <height>28</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>180</width>
     <height>0</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>180</width>
     <height>16777215</height>
    </size>
   </property>
   <property name="text">
    <string>Add restrict function</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="all_restrict_functions">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>480</y>
     <width>300</width>
     <height>150</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>300</width>
     <height>150</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>300</width>
     <height>150</height>
    </size>
   </property>
  </widget>
  <widget class="QTextEdit" name="all_linear_bounds">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>280</y>
     <width>300</width>
     <height>60</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>300</width>
     <height>60</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>300</width>
     <height>60</height>
    </size>
   </property>
  </widget>
  <widget class="QPushButton" name="insert_linear_bounds">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>240</y>
     <width>180</width>
     <height>28</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>180</width>
     <height>0</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>180</width>
     <height>16777215</height>
    </size>
   </property>
   <property name="text">
    <string>Insert linear bounds</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="line_insert_linear_bounds">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>200</y>
     <width>300</width>
     <height>30</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>300</width>
     <height>30</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>300</width>
     <height>30</height>
    </size>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>60</x>
     <y>160</y>
     <width>280</width>
     <height>25</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>280</width>
     <height>25</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>280</width>
     <height>25</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>[min_n, max_n], [] for default [-5, 5]</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>130</y>
     <width>310</width>
     <height>25</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>310</width>
     <height>25</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>310</width>
     <height>25</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Insert linear bounds for each variable as:</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>10</y>
     <width>180</width>
     <height>25</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>180</width>
     <height>25</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>180</width>
     <height>25</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Insert objective function</string>
   </property>
  </widget>
  <widget class="QPushButton" name="clear_linear_bounds">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>240</y>
     <width>180</width>
     <height>28</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>180</width>
     <height>0</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>180</width>
     <height>16777215</height>
    </size>
   </property>
   <property name="text">
    <string>Clear all linear bounds</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>480</x>
     <y>130</y>
     <width>160</width>
     <height>25</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>160</width>
     <height>25</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>160</width>
     <height>25</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Number of iterations</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>130</y>
     <width>120</width>
     <height>25</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>120</width>
     <height>25</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>80</width>
     <height>25</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Variable F [0, 2]</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_8">
   <property name="geometry">
    <rect>
     <x>660</x>
     <y>220</y>
     <width>140</width>
     <height>25</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>140</width>
     <height>25</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>140</width>
     <height>25</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Variable CR [0, 1]</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="CR_var">
   <property name="geometry">
    <rect>
     <x>680</x>
     <y>260</y>
     <width>100</width>
     <height>30</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>100</width>
     <height>30</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>100</width>
     <height>30</height>
    </size>
   </property>
  </widget>
  <widget class="QLabel" name="label_9">
   <property name="geometry">
    <rect>
     <x>480</x>
     <y>220</y>
     <width>160</width>
     <height>25</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>160</width>
     <height>25</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>160</width>
     <height>25</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Population number</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="number_arg">
   <property name="geometry">
    <rect>
     <x>490</x>
     <y>80</y>
     <width>51</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="value">
    <number>2</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_10">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>80</y>
     <width>180</width>
     <height>25</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>180</width>
     <height>25</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>180</width>
     <height>25</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Number of arguments</string>
   </property>
  </widget>
  <widget class="QPushButton" name="calculate">
   <property name="geometry">
    <rect>
     <x>550</x>
     <y>310</y>
     <width>180</width>
     <height>50</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>180</width>
     <height>50</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>180</width>
     <height>50</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Calculate</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="min_value">
   <property name="geometry">
    <rect>
     <x>540</x>
     <y>510</y>
     <width>250</width>
     <height>30</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>250</width>
     <height>30</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>250</width>
     <height>30</height>
    </size>
   </property>
  </widget>
  <widget class="QTextEdit" name="min_arguments">
   <property name="geometry">
    <rect>
     <x>540</x>
     <y>600</y>
     <width>350</width>
     <height>30</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>350</width>
     <height>30</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>350</width>
     <height>30</height>
    </size>
   </property>
  </widget>
  <widget class="QLabel" name="label_11">
   <property name="geometry">
    <rect>
     <x>540</x>
     <y>480</y>
     <width>125</width>
     <height>25</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>125</width>
     <height>25</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>125</width>
     <height>25</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Minimum value</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_12">
   <property name="geometry">
    <rect>
     <x>540</x>
     <y>570</y>
     <width>140</width>
     <height>25</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>140</width>
     <height>25</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>140</width>
     <height>25</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Minimum element</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="iteration_number">
   <property name="geometry">
    <rect>
     <x>500</x>
     <y>180</y>
     <width>100</width>
     <height>30</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>100</width>
     <height>30</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>100</width>
     <height>30</height>
    </size>
   </property>
  </widget>
  <widget class="QLineEdit" name="population_number">
   <property name="geometry">
    <rect>
     <x>500</x>
     <y>260</y>
     <width>100</width>
     <height>30</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>100</width>
     <height>30</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>100</width>
     <height>30</height>
    </size>
   </property>
  </widget>
  <widget class="QLineEdit" name="F_var">
   <property name="geometry">
    <rect>
     <x>680</x>
     <y>180</y>
     <width>100</width>
     <height>30</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>100</width>
     <height>30</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>100</width>
     <height>30</height>
    </size>
   </property>
  </widget>
  <widget class="QPushButton" name="plot">
   <property name="geometry">
    <rect>
     <x>550</x>
     <y>380</y>
     <width>180</width>
     <height>50</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>180</width>
     <height>50</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>180</width>
     <height>50</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Plot</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="trail">
   <property name="geometry">
    <rect>
     <x>740</x>
     <y>400</y>
     <width>151</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Show search trail</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="all_restrict_functions_values">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>480</y>
     <width>150</width>
     <height>150</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>150</width>
     <height>150</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>150</width>
     <height>150</height>
    </size>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>clear_restrict_functions</sender>
   <signal>clicked()</signal>
   <receiver>all_restrict_functions</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>280</x>
     <y>455</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>509</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clear_linear_bounds</sender>
   <signal>clicked()</signal>
   <receiver>all_linear_bounds</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>309</x>
     <y>240</y>
    </hint>
    <hint type="destinationlabel">
     <x>238</x>
     <y>282</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clear_restrict_functions</sender>
   <signal>clicked()</signal>
   <receiver>all_restrict_functions_values</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>311</x>
     <y>452</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>517</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>calculate</sender>
   <signal>clicked()</signal>
   <receiver>all_restrict_functions_values</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>581</x>
     <y>333</y>
    </hint>
    <hint type="destinationlabel">
     <x>485</x>
     <y>505</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
